src/backends/web.js

```diff
@@ -29,670 +40,1035 @@ async function dvui_open_file_picker(accept, multiple) {
+                }`,
+            );
             return null;
         }
 
-        programInfo = {
+        this.programInfo = {
             attribLocations: {
-                vertexPosition: gl.getAttribLocation(shaderProgram, "aVertexPosition"),
-                vertexColor: gl.getAttribLocation(shaderProgram, "aVertexColor"),
-                textureCoord: gl.getAttribLocation(shaderProgram, "aTextureCoord"),
+                vertexPosition: this.gl.getAttribLocation(
+                    this.shaderProgram,
+                    "aVertexPosition",
+                ),
+                vertexColor: this.gl.getAttribLocation(
+                    this.shaderProgram,
+                    "aVertexColor",
+                ),
+                textureCoord: this.gl.getAttribLocation(
+                    this.shaderProgram,
+                    "aTextureCoord",
+                ),
             },
             uniformLocations: {
-                matrix: gl.getUniformLocation(shaderProgram, "uMatrix"),
-                uSampler: gl.getUniformLocation(shaderProgram, "uSampler"),
-                useTex: gl.getUniformLocation(shaderProgram, "useTex"),
+                matrix: this.gl.getUniformLocation(
+                    this.shaderProgram,
+                    "uMatrix",
+                ),
+                uSampler: this.gl.getUniformLocation(
+                    this.shaderProgram,
+                    "uSampler",
+                ),
+                useTex: this.gl.getUniformLocation(
+                    this.shaderProgram,
+                    "useTex",
+                ),
             },
         };
 
-        indexBuffer = gl.createBuffer();
-        vertexBuffer = gl.createBuffer();
+        this.indexBuffer = this.gl.createBuffer();
+        this.vertexBuffer = this.gl.createBuffer();
+
+        this.gl.enable(this.gl.BLEND);
+        this.gl.blendFunc(this.gl.ONE, this.gl.ONE_MINUS_SRC_ALPHA);
+        this.gl.enable(this.gl.SCISSOR_TEST);
+        this.gl.scissor(
+            0,
+            0,
+            this.gl.canvas.clientWidth,
+            this.gl.canvas.clientHeight,
+        );
+    }
+
+    init() {
+        let app_init_return = 0;
+        let str = utf8encoder.encode(navigator.platform);
+        if (str.length > 0) {
+            const ptr = this.instance.exports.gpa_u8(
+                str.length,
+            );
+            var dest = new Uint8Array(
+                this.instance.exports.memory.buffer,
+                ptr,
+                str.length,
+            );
+            dest.set(str);
+            app_init_return = this.instance.exports.app_init(
+                ptr,
+                str.length,
+            );
+            this.instance.exports.gpa_free(ptr, str.length);
+        } else {
+            app_init_return = this.instance.exports.app_init(
+                0,
+                0,
+            );
+        }
+
+        if (app_init_return != 0) {
+            console.log(
```

Comment:
minor nitpick
```suggestion
            console.error(
```
maybe even consider throwing/returning an error code for people to handle
but maybe you didn't write this part

Reply:
This remained from the previous code which has just been moved. I'm debating if the user should call this `init` function or if it's only going to be called from `run`. In either case it is probably better to throw an error so that is can be handled or shows up in the console

Reply:
I would vote you throw the error, but probably want to start doing that elsewhere here too so you can leave it for now if you like... when I migrate my project's SDK to this new version, I will give it more thought